pandas/tests/frame/methods/test_quantile.py

```diff
@@ -14,6 +14,19 @@
 
 
 class TestDataFrameQuantile:
+    def test_numeric_only_default_false_warning(self):
+        # GH #7308
+        df = DataFrame({"A": [1, 2, 3], "B": [2, 3, 4]})
```

Comment:
Could you also test with string, object columns etc. Everything which should get caught by the no numeric check